subsys/bluetooth/gatt_dm.c

```diff
@@ -311,6 +315,7 @@ static uint8_t discovery_process_service(struct bt_gatt_dm *dm,
 
 	if (cur_attr->handle == cur_service_val->end_handle) {
 		/* No characteristics to discover, go to next service. */
+		discovery_complete(dm);
```

Comment:
Wouldn't be better to call this callback in next iteration when no other service with the same UUID is found ?

Reply:
Can you provide the scenario when this method could fail?

Reply:
There can be corner case situation when there are two the same service instances and the first one is empty, then would be good to discover a second one, but this scenario is very unlikely.